src/openforms/forms/api/viewsets.py

```diff
@@ -517,6 +518,17 @@ def variables_bulk_update(self, request, *args, **kwargs):
     def variables_list(self, request, *args, **kwargs):
         form = self.get_object()
         form_variables = form.formvariable_set.all()
+        if self.request.query_params:
+            filter_serializer = FormVariableSourceSerializer(
+                data=self.request.query_params
+            )
+            is_valid = filter_serializer.is_valid()
+            if not is_valid:
+                return Response([], status=status.HTTP_200_OK)
+
+            form_variables = form_variables.filter(
+                source=filter_serializer.validated_data["source"]
+            )
```

Comment:
wouldn't it make more sense to use https://pypi.org/project/django-filter/? Afaik it also has API schema introspection capabilities in combination with drf-spectacular.

Reply:
Yes I wanted to use it, but I couldn't quite figure out how to make it work for a viewset custom action (since the variables endpoint is an action in the form viewset).

Reply:
I think you can leverage the `filter_backends` keyword argument, see https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing